python/pyspark/sql/functions.py

```diff
@@ -136,6 +136,7 @@ def _():
                  'measured in radians.',
 
     'bitwiseNOT': 'Computes bitwise not.',
+    'bin': 'Computes the binary format of the given value.',
```

Comment:
let's just define a bin function in python, like other functions, and define a docstring test for it. don't forget to add it to __all__ on the top.